x-pack/legacy/plugins/ml/public/jobs/new_job_new/common/job_creator/job_creator.ts

```diff
@@ -230,6 +235,48 @@ export class JobCreator {
     }
   }
 
+  public get sparseData(): boolean {
+    return this._sparseData;
+  }
+
+  public set sparseData(sparseData: boolean) {
+    this._sparseData = sparseData;
+    this._updateSparseDataDetectors();
+  }
+
+  private _updateSparseDataDetectors() {
+    this._detectors.forEach((d, i) => {
```

Comment:
nit: Using `.map()` and reassigning to `this._detectors` might be a bit more verbose but it would make it more obvious that the code is transforming `this._detectors` itself. with `forEach()` and changing the references the code is less "pure".

Reply:
`this._detectors` is a reference to `this._job_config.analysis_config.detectors` and so can't be reassigned to a new array.
the job creator class has a couple of deep references like this to parts of the job config.